test/fw/ptl.csh

```diff
@@ -37,15 +37,43 @@
 
 # This file will set path variables in case of ptl installation
 
-if [ -f "/etc/debian_version" ]; then
-	ptl_prefix_lib=$( dpkg -L pbspro-ptl 2>/dev/null | grep -m 1 lib$ 2>/dev/null )
+if ( -f "/etc/debian_version" ) then
+	set ptl_prefix_lib=`dpkg -L pbspro-ptl 2>/dev/null | grep -m 1 lib$ 2>/dev/null`
 else
-	ptl_prefix_lib=$( rpm -ql pbspro-ptl 2>/dev/null | grep -m 1 lib$ 2>/dev/null )
-fi
-if [ "x${ptl_prefix_lib}" != "x" ]; then
-	python_dir=$( /bin/ls -1 ${ptl_prefix_lib} )
-	prefix=$( dirname ${ptl_prefix_lib} )
+	set ptl_prefix_lib=`rpm -ql pbspro-ptl 2>/dev/null | grep -m 1 lib$ 2>/dev/null`
+endif
+if ( ! $?ptl_prefix_lib ) then
+	set python_dir=`/bin/ls -1 ${ptl_prefix_lib}`
+	set prefix=`dirname ${ptl_prefix_lib}`
 
 	setenv PATH=${prefix}/bin/:${PATH} 
-	setenv PYTHONPATH=${prefix}/lib/${python_dir}/site-packages/:$PYTHONPATH 
-fi
+	setenv PYTHONPATH=${prefix}/lib/${python_dir}/site-packages/:$PYTHONPATH
+	unset python_dir
+	unset prefix
+	unset ptl_prefix_lib
+else
+	if ( $?PBS_CONF_FILE ) then
+		set conf = "$PBS_CONF_FILE"
+	else
+		set conf = /etc/pbs.conf
+	endif
+	if ( -r "${conf}" ) then
+		# we are not sourcing pbs.conf file because we don't want to export any variable, just parse PBS_EXEC
```

Comment:
why don't we do just that, source the file and use only PBS_EXEC?

Reply:
That would create a bunch of sourced variables in the environment. Every time the user logs in the file will be sourced which may or may not cause problems. We want not to tamper too much with the shell environment and only set the variables that we require.

Reply:
I don't think there will be any harm if the variables are set in the environment, but I am alright with the current logic.
The comment is too verbose, let us shorten it.